filters/united-states/arizona/tempe/public-art-geojson-to-json.js

```diff
@@ -1,51 +1,86 @@
-const filter = function (data, std_lib, stringify, skip_errors) {
+const filter = function (data, std_lib, schema, validator, stringify) {
+    // check for standard library and pull out required functions
+    if (!std_lib) {
+        throw "standard library not provided";
+    }
+    let add_required = std_lib.get("add_required");
+    let add_if_not_null = std_lib.get("add_if_not_null")
+    let remove_if_null = std_lib.get("remove_if_null");
+    let remove_if_empty = std_lib.get("remove_if_empty");
+    let validate_params = std_lib.get("validate_params");
+    let remove_null_date_fields = std_lib.get("remove_null_date_fields");
+    let create_dates_template = std_lib.get("create_dates_template");
+    
+    // validate parameters object
+    schema = validate_params(schema, validator);
+
     if (typeof data === 'string' || data instanceof String) {
         data = JSON.parse(data);
     }
     
     let new_data = [];
+    let errors = [];
 
-    data.features?.map(data => {
-        let d = data.properties || {}
+    data.features?.map(d => {
         let item = {};
+        let skip = false;
         
-        item.name = d.Name;
-        if (item.name === undefined) {
-            console.log(`Data name not found for art with ID ${d.OBJECTID}`);
+        // add name (required)
+        if (!add_required(item, "name", d, d.properties.Name, errors)) {
+            skip = true;
         }
 
-        let coordinates = { longitude: d.Longitude, latitude: d.Latitude };
-        if (coordinates.longitude === undefined || coordinates.latitude === undefined) {
-            console.log(`Data coordinates not found for art with ID ${d.OBJECTID}`);
+        // add coordinates (required)
+        let coordinates = {};
+        if (!add_required(coordinates, "longitude", d, d.properties.Longitude, errors)) {
+            skip = true;
+        }
+        if (!add_required(coordinates, "latitude", d, d.properties.Latitude, errors)) {
+            skip = true;
         }
         item.coordinates = coordinates;
 
-        item.details = {
-          artist: d.Artist,
-          year: d.Year,
-          address: d.Address,
-          hours: d.Hours,
-          tab_name: d.Tab_Name,
-          short_desc: d.Short_Desc,
-          desc1: d.Desc1,
-          desc2: d.Desc2,
-          desc3: d.Desc3,
-          desc4: d.Desc4,
-          desc5: d.Desc5,
-          neighborhood: d.Neighborhood,
-          developer: d.Developer,
-          website: d.Website,
-          pic_url: d.Pic_URL,
-        }
+        //optional fields
+        // item.artist = d.properties.ARTIST_FIRST_NAME + " " + d.properties.ARTIST_LAST_NAME
+        add_if_not_null(item, "artist", d.properties.Artist);
+        add_if_not_null(item, "material", d.properties.Hours);
+        add_if_not_null(item, "area", d.properties.Neighborhood);
+        add_if_not_null(item, "image_url", [d.properties.Pic_URL]);
```

Comment:
change to the name image_urls to conform to schema, and add remove_if_empty(item, "image_urls"). Also I don't think you need add_if_not_null here, since it won't do anything since you are adding an array to the field

Reply:
correct, the array will always be defined even if empty. Maybe a better way to do this is with an if statement to check if Pic_URL is blank, undefined, or null ( if (d.properties.Pic_URL) {} ) and then if not blank add it to the array. Then as Sarah said add a remove_if_empty call.